src/crawlers/browser_crawler.js

```diff
@@ -360,8 +373,12 @@ export default class BrowserCrawler extends BasicCrawler {
      * @internal
      */
     async _handleRequestFunction(crawlingContext) {
-        const { id } = crawlingContext;
-        const page = await this.browserPool.newPage({ id });
+        const newPageOptions = {
+            id: crawlingContext.id,
+            pageOptions: this._getNewPageOptions(crawlingContext),
+        };
```

Comment:
I would include even the `id` to make it look like this:

```js
const newPageOptions = this._getNewPageOptions(crawlingContext);
const page = await this.browserPool.newPage(newPageOptions);
```
What do you think?